File: ILEYA/DAY 1/level 1/PYTHON/PalindromeCheck.py
def is_palindromic(nums):
left  = 0
right = len(nums) - 1


while left < right:
    if nums[left] != nums[right]:
        return False
    left  += 1
    right -= 1

return True


if name == “main”:
palindrome     = [45, 0, 8, 0, 45]
not_palindrome = [1, 2, 3]


print(f"Input : {palindrome}")
print(f"Result: {is_palindromic(palindrome)}")     

print(f"Input : {not_palindrome}")
print(f"Result: {is_palindromic(not_palindrome)}") 

File: ILEYA/DAY 1/level 1/JAVA/PalindromeCheck.java
import java.util.Arrays;

public class PalindromeCheck {


public static boolean isPalindromic(int[] nums) {
    int left  = 0;
    int right = nums.length - 1;

    while (left < right) {
        if (nums[left] != nums[right]) {
            return false;
        }
        left++;
        right--;
    }
    return true;
}

public static void main(String[] args) {
    int[] palindrome    = { 45, 0, 8, 0, 45 };
    int[] notPalindrome = { 1, 2, 3 };

    System.out.println("Input : " + Arrays.toString(palindrome));
    System.out.println("Result: " + isPalindromic(palindrome));   

    System.out.println("Input : " + Arrays.toString(notPalindrome));
    System.out.println("Result: " + isPalindromic(notPalindrome)); 
}


}
